src/ShowingTransitionSupport.js

```diff
@@ -0,0 +1,242 @@
+	*/
+	showingClass: undefined,
+
+	/**
+	* Initializes the defaults, and prepares the component with its classes in case of initially
+	* "showing: false".
+	*
+	* @method
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			sup.apply(this, arguments);
+
+			this.showingDuration = (this.showingDuration === undefined) ? null      : this.showingDuration;
+			this.hidingDuration  = (this.hidingDuration  === undefined) ? null      : this.hidingDuration;
+			this.shownMethod     = (this.shownMethod     === undefined) ? null      : this.shownMethod;
+			this.hiddenMethod    = (this.hiddenMethod    === undefined) ? null      : this.hiddenMethod;
+			this.shownClass      = (this.shownClass      === undefined) ? 'shown'   : this.shownClass;
+			this.hiddenClass     = (this.hiddenClass     === undefined) ? 'hidden'  : this.hiddenClass;
+			this.hidingClass     = (this.hidingClass     === undefined) ? 'hiding'  : this.hidingClass;
+			this.showingClass    = (this.showingClass    === undefined) ? 'showing' : this.showingClass;
+			if (this.shownMethod) this.shownMethodChanged();
+			if (this.hiddenMethod) this.hiddenMethodChanged();
+			this.showingChanged();
+		};
+	}),
+
+	/**
+	* Overrides the showingChanged handler to add support for transitions at the right times and
+	* places.
+	*
+	* @method
+	* @private
+	*/
+	showingChanged: kind.inherit(function (sup) {
+		return function (sender, ev) {
+			var args = arguments;
+
+			// Prepare our visual state
+			this.applyStyle('display', null);
+			this.applyStyle('visibility', null);
+			if (this.showing) {
+				// Reset our state classes, in case we switched mid-stream
+				this.removeClass(this.hidingClass);
+				this.removeClass(this.hiddenClass);
+				sup.apply(this, args);
+				if (this.showingDuration && this.hasNode()) {
+					this.set('showingTransitioning', true);
+					// Start transition: Apply a class and start a timer.
+					// When timer finishes, run the exit function,
+					// remove the transitioning class
+					// and add the final-state class
+					this.addClass(this.showingClass);
+					this.startJob('showingTransition', function () {
+						utils.call(this._shownMethodScope, this.shownMethod);	// Run the supplied method.
+						this.removeClass(this.showingClass);
+						this.addClass(this.shownClass);
+						this.set('showingTransitioning', false);
+					}, this.showingDuration);
+				} else {
+					// No transition, just a shown class.
+					this.stopJob('showingTransition');
+					this.addClass(this.shownClass);
+				}
+			} else {
+				// Reset our state classes, in case we switched mid-stream
+				this.removeClass(this.showingClass);
+				this.removeClass(this.shownClass);
+				if (this.hidingDuration && this.hasNode()) {
+					this.set('showingTransitioning', true);
+					this.addClass(this.hidingClass);
+					this.startJob('showingTransition', function () {
+						utils.call(this._hiddenMethodScope, this.hiddenMethod);	// Run the supplied method.
```

Comment:
I think the callback should fire when hiding/showing duration is 0/undefined as well. In cases where the duration would be configurable external to the control mixing ShowingTransitionSupport but the control relies on the callback to complete actions, it would need it in both cases. WDYT?